relay-kafka/src/config.rs

```diff
@@ -80,93 +80,62 @@ impl KafkaTopic {
-        match kafka_topic {
-            KafkaTopic::Attachments => &self.attachments,
-            KafkaTopic::Events => &self.events,
-            KafkaTopic::Transactions => &self.transactions,
-            KafkaTopic::Outcomes => &self.outcomes,
-            KafkaTopic::OutcomesBilling => self.outcomes_billing.as_ref().unwrap_or(&self.outcomes),
-            KafkaTopic::MetricsSessions => &self.metrics_sessions,
-            KafkaTopic::MetricsGeneric => &self.metrics_generic,
-            KafkaTopic::Profiles => &self.profiles,
-            KafkaTopic::ReplayEvents => &self.replay_events,
-            KafkaTopic::ReplayRecordings => &self.replay_recordings,
-            KafkaTopic::Monitors => &self.monitors,
-            KafkaTopic::Spans => &self.spans,
-            KafkaTopic::MetricsSummaries => &self.metrics_summaries,
-            KafkaTopic::Cogs => &self.cogs,
+        /// Configuration for topics.
+        #[derive(Serialize, Deserialize, Debug)]
+        #[serde(default)]
+        pub struct TopicAssignments {
+            $(
+                /// Topic name for $default_topic
+                #[serde(alias = $default_topic)]
+                pub $field_name: TopicAssignment,
+            )*
         }
-    }
-}
 
-impl Default for TopicAssignments {
-    fn default() -> Self {
-        Self {
-            events: "ingest-events".to_owned().into(),
-            attachments: "ingest-attachments".to_owned().into(),
-            transactions: "ingest-transactions".to_owned().into(),
-            outcomes: "outcomes".to_owned().into(),
-            outcomes_billing: None,
-            metrics_sessions: "ingest-metrics".to_owned().into(),
-            metrics_generic: "ingest-performance-metrics".to_owned().into(),
-            profiles: "profiles".to_owned().into(),
-            replay_events: "ingest-replay-events".to_owned().into(),
-            replay_recordings: "ingest-replay-recordings".to_owned().into(),
-            monitors: "ingest-monitors".to_owned().into(),
-            spans: "snuba-spans".to_owned().into(),
-            metrics_summaries: "snuba-metrics-summaries".to_owned().into(),
-            cogs: "shared-resources-usage".to_owned().into(),
+        impl TopicAssignments{
+            /// Get a topic assignment by [`KafkaTopic`] value
+            #[must_use]
+            pub fn get(&self, kafka_topic: KafkaTopic) -> &TopicAssignment {
+                match kafka_topic {
+                    $(
+                        $kafka_topic => &self.$field_name,
+                    )*
+                }
+            }
+        }
+
+        impl Default for TopicAssignments {
+            fn default() -> Self {
+                Self {
+                    $(
+                        $field_name: $default_topic.to_owned().into(),
+                    )*
+
+                }
+            }
         }
+    };
+}
+
+define_topic_assignments! {
+    TopicAssignments {
```

Comment:
Can we get the documentation back? It should be possible to include it in the macro and just generate it on the fields using `#[doc = $foo]`.

Reply:
Sounds good, added back the original descriptions